build.gradle

```diff
@@ -22,6 +22,17 @@ dependencies {
 
 mainClassName = 'com.gradle.exportapi.Application'
 
+task addAuthorization {
+    description 'Adds authorization to the system properties to be available at runtime if configured'
+    onlyIf {
+        System.getProperty("basic_auth") == null && System.getProperty("password") != null
+    }
+    doLast {
+        def username = System.getProperty("username")
+        tasks.run.systemProperties.put("basic_auth", "$username:${System.getProperty("password")}".bytes.encodeBase64().toString())
```

Comment:
Is it generally regarded as being consistent with gradle idiom and best practice to have a task configure a subsequent task during execution? I have the impression that there's momentum toward making tasks and similar objects immutable after the configuration phase?

I would probably put this logic in a `doFirst{}` on the `run` task itself.